conda_lock/pypi_solver.py

```diff
@@ -373,6 +375,9 @@ def _prepare_repositories_pool(allow_pypi_requests: bool) -> Pool:
     factory = Factory()
     config = factory.create_config()
     repos = [
+        factory.create_legacy_repository({"name": name, "url": url}, config)
+        for name, url in _parse_repositories_from_environment().items()
+    ] + [
```

Comment:
Could you please write a test for `_prepare_repositories_pool` to exercise this functionality?